modules/core/map/js/farm_map.js

```diff
@@ -2,62 +2,17 @@
   Drupal.behaviors.farm_map = {
     attach: function (context, settings) {
 
-      // Get the units.
-      let units = 'metric';
-      if (!!drupalSettings.farm_map.units) {
-        units = drupalSettings.farm_map.units;
-      }
-
-      // Build default options.
-      const defaultOptions = {
-        units,
-        interactions: {
-          onFocusOnly: true
-        },
-      };
       context.querySelectorAll('.farm-map').forEach(function (element) {
 
-        // Only create a map once per element.
-        if (element.getAttribute('processed')) return;
-        element.setAttribute('processed', true);
-
-        element.setAttribute('tabIndex', 0);
-        const mapId = element.getAttribute('id');
-        const mapOptions = { ...defaultOptions, ...drupalSettings.farm_map[mapId].instance};
-        const instance = farmOS.map.create(mapId, mapOptions);
-        context.querySelectorAll('.ol-popup-closer').forEach(function (element) {
-          element.onClick = function (element) {
-            element.focus();
-          };
-        });
+          // Only create a map once per element.
+          if (element.getAttribute('processed')) return;
+          element.setAttribute('processed', true);
 
-        // If the map is rendered as part of a form field, update the map size
-        // when the field's visible state changes,
-        const formWrapper = element.closest('div.form-wrapper');
-        if (formWrapper != null) {
-          const formWrapperObserver = new MutationObserver((mutations) => {
-
-            // Only update the map size if the wrapper was previously hidden.
-            if (mutations.some((mutation) => { return mutation.oldValue.includes('display: none')})) {
-              instance.map.updateSize();
-            }
-          });
-
-          // Observe the style attribute.
-          formWrapperObserver.observe(formWrapper, {
-            attributeFilter: ["style"],
-            attributeOldValue: true
-          })
-        }
-
-        // If the map is inside a details element, update the map size when
-        // the details element is toggled.
-        const details = element.closest('details');
-        if (details != null) {
-          details.addEventListener('toggle', function() {
-            instance.map.updateSize();
-          });
-        }
+          element.setAttribute('tabIndex', 0);
+
+          const mapId = element.getAttribute('id');
+          const mapInstanceOptions = {};
+          Drupal.behaviors.farm_map.createMapInstance(context, element, mapId, mapInstanceOptions);
```

Comment:
Minor, but I think we could change this line to be:

```js
this.createMapInstance(context, element, mapId, mapInstanceOptions);
```

Reply:
I think that's true, will test that if we stick with keeping the factory method in this JS library...